pyteal/ast/subroutine.py

```diff
@@ -78,21 +100,26 @@ def name(self) -> str:
     def argumentCount(self) -> int:
         return len(self.implementationParams)
 
-    def invoke(self, args: List[Expr]) -> "SubroutineCall":
+    def arguments(self) -> List[str]:
+        return list(self.implementationParams.keys())
+
+    def invoke(self, args: List[Union[Expr, ScratchVar]]) -> "SubroutineCall":
         if len(args) != self.argumentCount():
             raise TealInputError(
                 "Incorrect number of arguments for subroutine call. Expected {} arguments, got {}".format(
                     self.argumentCount(), len(args)
                 )
             )
 
-        for i, arg in enumerate(args):
-            if not isinstance(arg, Expr):
-                raise TealInputError(
-                    "Argument at index {} of subroutine call is not a PyTeal expression: {}".format(
-                        i, arg
+        if self.compiler_version is not None and self.compiler_version >= 6:
+            for i, arg in enumerate(args):
+                atype = self.expected_arg_types[i]
+                if not isinstance(arg, atype):
```

Comment:
was failing some older compiler tests but maybe it's more because of the issue you joined pointed out... in general, if people annotated their subroutines in the past our new annotation restrictions break their old code.

Reply:
I've removed all the compiler version gating